src/ray/raylet/main.cc

```diff
@@ -168,22 +169,9 @@ int main(int argc, char *argv[]) {
   auto handler = [&main_service, &raylet_socket_name, &server, &gcs_client](
                      const boost::system::error_code &error, int signal_number) {
     RAY_LOG(INFO) << "Raylet received SIGTERM, shutting down...";
-    auto shutdown_callback = [&server, &main_service, &gcs_client]() {
-      server.reset();
-      gcs_client->Disconnect();
-      main_service.stop();
-    };
-    RAY_CHECK_OK(gcs_client->client_table().Disconnect(shutdown_callback));
-    // Give a timeout for this Disconnect operation.
-    boost::posix_time::milliseconds stop_timeout(800);
-    boost::asio::deadline_timer timer(main_service);
-    timer.expires_from_now(stop_timeout);
-    timer.async_wait([shutdown_callback](const boost::system::error_code &error) {
-      if (!error) {
-        RAY_LOG(INFO) << "Disconnect from client table timed out, forcing shutdown.";
-        shutdown_callback();
-      }
-    });
```

Comment:
is it ok to return this?

Reply:
Yes. The old logic between line 178 to line 186 is mark self down to GCS with a `async method` `client_table().Disconnect(...)`, then  wait a timeout to exit.
 After we replace the `async method` with `sync method` (move it into method raylet::Stop()),  it looks brief.

Reply:
This is indeed nice!   #6272 contains two fixes, the one related to client table disconnection can be reverted.